docs/aws/audit/esmonitoring/rules/es_ri_payment_pending.mdx

```diff
@@ -23,6 +23,253 @@ CBP
+1. Install and configure AWS CLI: Before you can start, you need to install the AWS CLI on your local machine. You can do this by downloading the appropriate installer from the AWS CLI website. Once installed, you can configure it by running `aws configure` and providing your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all Elasticsearch Reserved Instances: Use the `es:describe-reserved-elasticsearch-instances` command to list all the Elasticsearch Reserved Instances. The command is as follows:
+
+   ```
+   aws es describe-reserved-elasticsearch-instances
+   ```
+
+3. Filter instances with Payment Pending status: The output of the above command will be a JSON object containing details about all the Elasticsearch Reserved Instances. You need to parse this JSON object to filter out instances that have the status 'payment-pending'. You can use the `jq` command-line JSON processor for this. The command is as follows:
+
+   ```
+   aws es describe-reserved-elasticsearch-instances | jq '.ReservedElasticsearchInstances[] | select(.State=="payment-pending")'
+   ```
+
+4. Check the output: If the above command returns any instances, it means that there are Elasticsearch Reserved Instances with Payment Pending status. If it doesn't return anything, it means that there are no such instances.
+</Accordion>
+
+<Accordion title='Using Python'>
+To detect if an Elasticsearch Reserved Instance has a status of "Payment Pending" using Python scripts, you can use the Boto3 library, which allows you to directly interact with AWS services, including Elasticsearch. Here are the steps:
+
+1. **Install and Configure Boto3:**
+   First, you need to install and configure Boto3 if you haven't done so already. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
+   ```
+
+2. **Initialize the Elasticsearch Client:**
+   Once Boto3 is installed and configured, you can initialize the Elasticsearch client in your Python script:
+   ```python
+   import boto3
+
+   es = boto3.client('es')
+   ```
+
+3. **List all Elasticsearch Reserved Instances:**
+   You can then use the `describe_reserved_elasticsearch_instances` method to get a list of all Elasticsearch Reserved Instances:
+   ```python
+   response = es.describe_reserved_elasticsearch_instances()
+   ```
+
+4. **Check the Status of Each Instance:**
+   Finally, you can iterate over the list of instances and check the `State` of each one. If the `State` is "payment-pending", you've detected a misconfiguration:
+   ```python
+   for instance in response['ReservedElasticsearchInstances']:
+       if instance['State'] == 'payment-pending':
+           print(f"Instance {instance['ReservedElasticsearchInstanceId']} has a status of Payment Pending.")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>